src/o365/spo/commands/mail/mail-send.ts

```diff
@@ -0,0 +1,250 @@
+import auth from '../../SpoAuth';
+import { ContextInfo } from '../../spo';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import * as request from 'request-promise-native';
+import {
+  CommandOption,
+  CommandValidate
+} from '../../../../Command';
+import SpoCommand from '../../SpoCommand';
+import Utils from '../../../../Utils';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  to: string;
+  subject: string;
+  body: string;
+  from?: string;
+  cc?: string;
+  bcc?: string;
+  additionalHeaders?: string;
+}
+
+interface TypedHash<T> {
+  [key: string]: T;
+}
+
+class SpoMailSendCommand extends SpoCommand {
+  public get name(): string {
+    return commands.MAIL_SEND;
+  }
+
+  public get description(): string {
+    return 'Send an email from SharePoint';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.to = typeof args.options.to !== 'undefined';
+    telemetryProps.subject = typeof args.options.subject !== 'undefined';
+    telemetryProps.body = typeof args.options.body !== 'undefined';
+    telemetryProps.from = typeof args.options.from !== 'undefined';
+    telemetryProps.cc = typeof args.options.cc !== 'undefined';
+    telemetryProps.bcc = typeof args.options.bcc !== 'undefined';
+    telemetryProps.additionalHeaders = typeof args.options.additionalHeaders !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    if (this.debug) {
+      cmd.log(`Retrieving access token for the site collection ${auth.service.resource}...`);
+    }
+
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((accessToken: string): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}. Retrieving request digest...`);
+        }
+
+        return this.getRequestDigest(cmd, this.debug);
+      })
+      .then((res: ContextInfo): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        const params = {
+          properties: this.extend({ "__metadata": { "type": "SP.Utilities.EmailProperties" } }, {
```

Comment:
Since this is an `any` object anyway, why use the complex `extend` structure rather than just assign properties to it?

Reply:
You're right! `extend` function is used by all of parameters and to keep a consistency in the code, I choose to use extend for each param. (same model for all)

Reply:
My point is that we're not using a similar construction anywhere else in the CLI and I'm yet to understand the benefit of it over just assigning properties?

Reply:
1. Decrease the number of code lines
2. Change only once for all params
3. The object can get complex with the headers param
I don't have any other idea ^^ 

It is the same point for all parameters, not only this one, isn't it ? If you prefer, I can try to remove it and for each object into properties directly as well.

Reply:
Just to double check we're talking about the same. You're using now:

```ts
        params.properties = this.extend(params.properties, {
          To: { results: args.options.to.replace(/\s+/g, '').split(',') },
        });
```

Why not use instead:

```ts
        params.properties.To = { results: args.options.to.replace(/\s+/g, '').split(',') };
```

The latter is what we do in all other commands in the CLI. To me, it is easier to read and consistent with what we have. If there is a drawback to doing this, we should discuss it separately and implement consistently across all commands.

Reply:
I understand
I remove it and check it

Reply:
Ok done! Know I'll spend time on the coverage :'(

Reply:
Thanks for such a quick turnaround 👍

Reply:
In the mean time, I saw that:

<img width="1779" alt="screenshot 2019-03-03 at 21 28 03" src="https://user-images.githubusercontent.com/15279205/53701262-68738e80-3dfb-11e9-97c5-8816fee53e7d.png">

<img width="849" alt="screenshot 2019-03-03 at 21 31 45" src="https://user-images.githubusercontent.com/15279205/53701295-d91aab00-3dfb-11e9-9aa1-ec37a3bec3a6.png">

Reply:
Done